front/app/modules/commercial/user_custom_fields/admin/containers/CustomFields/RegistrationCustomFieldEdit/RegistrationCustomFieldOptions.tsx

```diff
@@ -28,14 +30,14 @@ import { injectIntl } from 'utils/cl-intl';
 import { InjectedIntlProps } from 'react-intl';
 
 const RegistrationCustomFieldOptions = memo(
-  ({ intl: { formatMessage } }: InjectedIntlProps) => {
-    const { customField } = useUserCustomFieldOutletContext();
-
-    if (isNilOrError(customField)) return null;
+  ({
+    intl: { formatMessage },
+    params: { userCustomFieldId },
+  }: InjectedIntlProps & WithRouterProps) => {
+    if (isNilOrError(userCustomFieldId)) return null;
 
-    const userCustomFieldOptions = useUserCustomFieldOptions(customField.id);
+    const userCustomFieldOptions = useUserCustomFieldOptions(userCustomFieldId);
```

Comment:
We should imo allow ids that are undefined and deal with them in the resource hooks so we don't need the check on the id, but out of scope.